### ISSUE TITLE
Unclear Unit Test

### ISSUE THREAD

**Issue by ArneTR** [green-coding-berlin/green-metrics-tool#545]
Hey @dan-mm 

I do not understand this test: https://github.com/green-coding-berlin/green-metrics-tool/blob/c13d73c79cb363f32cb55566401f377f4405f1c6/tests/test_usage_scenario.py#L73

It references the wrong values in this file: https://github.com/green-coding-berlin/green-metrics-tool/blob/main/tests/data/usage_scenarios/env_vars_stress_unallowed.yml

How I understand the test it will only check if *any* of the variables will trigger a fail, but not everyone individually. Which is what we want, not?

**Comment by ArneTR** [comment]
Als I do not understand the next two tests in the file:

```
def test_env_variable_skip_unsafe_true():
    runner = Tests.setup_runner(usage_scenario='env_vars_stress.yml', skip_unsafe=True)
    env_var_output = get_env_vars(runner)
    assert env_var_output == 'alpha-num123_\n', Tests.assertion_info('alpha-num123_', env_var_output)

def test_env_variable_allow_unsafe_true():
    runner = Tests.setup_runner(usage_scenario='env_vars_stress.yml', allow_unsafe=True)
    env_var_output = get_env_vars(runner)
    assert env_var_output == 'alpha-num123_\n', Tests.assertion_info('alpha-num123_', env_var_output)
```

Why are you testing here with skip_unsafe if the variables are anyway safe to use? Shouldn't you test on the https://github.com/green-coding-berlin/green-metrics-tool/blob/main/tests/data/usage_scenarios/env_vars_stress_unallowed.yml file rather than the https://github.com/green-coding-berlin/green-metrics-tool/blob/main/tests/data/usage_scenarios/env_vars_stress.yml ?

**Comment by dan-mm** [comment]
for the latter two tests (test_env_variable_skip_unsafe_true && test_env_variable_allow_unsafe_true) you're right. I've gone through and tried to understand why I did it that way, but don't see the logic. I think it was just a mistake on my part. I see you've made the changes for these two tests in https://github.com/green-coding-berlin/green-metrics-tool/pull/546 - I've checked them out and it looks good.

For the first test (test_env_variable_no_skip_or_allow) - you're right that it only checks if ANY of the env vars are present trigger a fail. This is intentional though just to save time. This test is relatively slow (15 seconds), and since this is not an exhaustive list of what's unallowed anyways, just a sample (the full list is based on docker specification, right?) - I wrote this test as a single check to make sure this functionality exists. We can add two more tests to be a bit more comprehensive but that'll add about 30 seconds to the test time and I'm not sure the added value is there.
